### PR TITLE
SONAR-6933 Drop newDuplication() beta API

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: sonar-batch/src/main/java/org/sonar/batch/cpd/AbstractCpdEngine.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        if (clonePartCount > MAX_CLONE_PART_PER_GROUP) {
+          LOG.warn("Too many duplication references on file " + inputFile.relativePath() + " for block at line " + originBlock.getStartLine() + ". Keep only the first "
+            + MAX_CLONE_PART_PER_GROUP + " references.");
+          break;
+        }
+        blockBuilder.clear();
+        String componentKey = duplicate.getResourceId();
+        if (!component.key().equals(componentKey)) {
+          BatchComponent sameProjectComponent = batchComponentCache.get(componentKey);
+          blockBuilder.setOtherFileRef(sameProjectComponent.batchId());
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/cpd/AbstractCpdEngine.java]: BLOCKER NullPointerException might be thrown as 'sameProjectComponent' is nullable here [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2259)

**Comment by SonarTech** [comment]
SonarQube analysis reported 7 issues:
* BLOCKER 1 blocker
* MAJOR 6 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Remove this unused "MAX_CLONE_GROUP_PER_FILE" private field.](https://github.com/SonarSource/sonarqube/blob/78eeeb913ddf645cc9adc5fc68202af7235991e8/sonar-batch/src/main/java/org/sonar/batch/cpd/JavaCpdEngine.java#L72) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1068)
* MAJOR [Remove this unused "MAX_CLONE_PART_PER_GROUP" private field.](https://github.com/SonarSource/sonarqube/blob/78eeeb913ddf645cc9adc5fc68202af7235991e8/sonar-batch/src/main/java/org/sonar/batch/cpd/JavaCpdEngine.java#L73) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1068)
* MAJOR ["MAX_CLONE_GROUP_PER_FILE" is the name of a field in "AbstractCpdEngine".](https://github.com/SonarSource/sonarqube/blob/78eeeb913ddf645cc9adc5fc68202af7235991e8/sonar-batch/src/main/java/org/sonar/batch/cpd/JavaCpdEngine.java#L72) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2387)
* MAJOR ["MAX_CLONE_PART_PER_GROUP" is the name of a field in "AbstractCpdEngine".](https://github.com/SonarSource/sonarqube/blob/78eeeb913ddf645cc9adc5fc68202af7235991e8/sonar-batch/src/main/java/org/sonar/batch/cpd/JavaCpdEngine.java#L73) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2387)
* MAJOR [Protected method 'doBeforeStart(...)' is never used.](https://github.com/SonarSource/sonarqube/blob/78eeeb913ddf645cc9adc5fc68202af7235991e8/sonar-batch/src/main/java/org/sonar/batch/scan/ProjectScanContainer.java#L107) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AUnusedProtectedMethod)
* MAJOR [Protected method 'doAfterStart(...)' is never used.](https://github.com/SonarSource/sonarqube/blob/78eeeb913ddf645cc9adc5fc68202af7235991e8/sonar-batch/src/main/java/org/sonar/batch/scan/ProjectScanContainer.java#L223) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AUnusedProtectedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues:
* BLOCKER 1 blocker
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Protected method 'doBeforeStart(...)' is never used.](https://github.com/SonarSource/sonarqube/blob/87399585fa1cf8e91e3c0a7665d2e4235a561779/sonar-batch/src/main/java/org/sonar/batch/scan/ProjectScanContainer.java#L107) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AUnusedProtectedMethod)
* MAJOR [Protected method 'doAfterStart(...)' is never used.](https://github.com/SonarSource/sonarqube/blob/87399585fa1cf8e91e3c0a7665d2e4235a561779/sonar-batch/src/main/java/org/sonar/batch/scan/ProjectScanContainer.java#L223) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AUnusedProtectedMethod)

**Comment by dbmeneses** [comment]
LGTM

**Pull request merged**